AppCenter/AppCenterSDK+HiDrive.swift

```diff
@@ -0,0 +1,150 @@
+		MSCrashes.setEnabled(configuration.enableCrashes)
+		MSCrashes.setDelegate(shared)
+	}
+
+	/// Returns True, if and only if the Service got started and is enabled.
+	var isDistributionEnabled: Bool {
+		return MSDistribute.isEnabled()
+	}
+
+	/// Will enable or disable the Distribution Feature of AppCenter
+	/// While disabling works always.
+	/// For enabling the Setup (aka. start) - Method should be called before.
+	///
+	/// Flow in the App, for Apps that want to dynamically change that State:
+	/// - Call the Setup Method for with `isDistributionEnabled` set to true.
+	/// - Disable Distribution using `enableDistribution(enabled: false)`
+	/// - Enable Distribution at a later time using the same method
+	///
+	/// - Parameter enabled: Enable or Disable Distribtion
+	func enableDistribution(enabled: Bool = true) {
+		MSDistribute.setEnabled(enabled)
+	}
+
+	/// This will create a `fatalError` to crash the app.
+	static func performTestCrash() {
+
+		MSCrashes.generateTestCrash()
+	}
+}
+
+extension AppCenterSDK {
+
+	struct Configuration {
+
+		fileprivate var enableDebug				: Bool
+		fileprivate var appSecret				: String
+		fileprivate var isDistributionEnabled	: Bool
+		fileprivate var enableCrashes			: Bool
+
+		/// Initializes a AppCenterSDK Configuration
+		///
+		///	- Parameters:
+		///		- appSecret: supply this manually if you dont want it in the info.plist
+		///		- enableDebug: Should start the Services even for Debug, Default is false
+		///		- isDistributionEnabled: Should start the Distribution Service, Default is false for Debug and Live apps, else true,
+		init(appSecret: String? = nil, enableDebug: Bool = false, isDistributionEnabled: Bool? = nil, enableCrashes: Bool = true) {
+
+			let appSecretFromBundle = Bundle.main.object(forInfoDictionaryKey: AppCenterConstants.appSecretKey) as? String
+
+			guard let _appSecret = (appSecret ?? appSecretFromBundle) else {
+				fatalError("You have to set the `\(AppCenterConstants.appSecretKey)` key in the info plist or specify your own when initializing the SDK.")
+			}
+
+			self.enableDebug			= enableDebug
+			self.appSecret				= _appSecret
+			self.enableCrashes			= enableCrashes
+
+			#if DEBUG
+			self.isDistributionEnabled	= isDistributionEnabled ?? false
+			#elseif LIVE
+			self.isDistributionEnabled	= isDistributionEnabled ?? false
+			#else
+			self.isDistributionEnabled	= isDistributionEnabled ?? true
+			#endif
+		}
+
+		static var `default`: AppCenterSDK.Configuration {
+			return Configuration()
+		}
+	}
+}
+
+extension AppCenterSDK: MSCrashesDelegate {
+
+	func attachments(with crashes: MSCrashes!, for errorReport: MSErrorReport!) -> [MSErrorAttachmentLog]! {
+		return [
+			MSErrorAttachmentLog.attachment(withText: self.applicationLog ?? "No Log found", filename: "SMFLogger.log")
+		]
+	}
+
+	private var applicationLog:  String? {
+		guard
+			let description = Logger.logFilesContent(maxSize: AppCenterConstants.smfLogUploadMaxSize),
```

Comment:
Here we should also use `isLogUploadEnabled` to define if we should return description or not